streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -600,6 +609,100 @@ private void to(final TopicNameExtractor<K, V> topicExtractor,
         builder.addGraphNode(streamsGraphNode, sinkNode);
     }
 
+    @Override
+    public KTable<K, V> toTable() {
+        return toTable(NamedInternal.empty());
+    }
+
+    @Override
+    public KTable<K, V> toTable(final Named named) {
+        final ConsumedInternal<K, V> consumedInternal = new ConsumedInternal<>(Consumed.with(keySerde, valSerde));
+
+        final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal =
+            new MaterializedInternal<>(
+                Materialized.with(consumedInternal.keySerde(), consumedInternal.valueSerde()),
+                builder,
+                TO_KTABLE_NAME);
+
+        return doToTable(named, materializedInternal);
+    }
+
+    @Override
+    public KTable<K, V> toTable(final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
+        return toTable(NamedInternal.empty(), materialized);
+    }
+
+    @Override
+    public KTable<K, V> toTable(final Named named,
+                                final Materialized<K, V, KeyValueStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(materialized, "materialized can't be null");
+
+        final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal =
+            new MaterializedInternal<>(materialized);
+        return doToTable(
+            named,
+            materializedInternal);
+    }
+
+    private KTable<K, V> doToTable(final Named named,
+                                   final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal) {
+        Objects.requireNonNull(named, "named can't be null");
+
+        final Serde<K> keySerde = materializedInternal.keySerde() == null
+            ? this.keySerde
+            : materializedInternal.keySerde();
+        final Serde<V> valueSerde = materializedInternal.valueSerde() == null
+            ? this.valSerde
+            : materializedInternal.valueSerde();
+
+        final NamedInternal namedInternal = new NamedInternal(named);
+        final String name = namedInternal.orElseGenerateWithPrefix(builder, TO_KTABLE_NAME);
+        final Set<String> nodes;
+        final StreamsGraphNode nextNode;
+
+        if (streamsGraphNode.isKeyChangingOperation()) {
```

Comment:
This seems to only verify the direct parent -- however, we should rather use `repartitionRequired` flag to determine if we need to insert a repartition topic or not.